scripts/monitor/visualization/resource_monitoring.py

```diff
@@ -0,0 +1,596 @@
+from typing import Dict
+import matplotlib.pyplot as plt
+from bokeh.models import Title
+from matplotlib.backends.backend_pdf import PdfPages
+import numpy
+import pandas as pd
+import pandas_bokeh
+from operator import truediv
+
+
+class ResourceUsage:
+    instances = 0  # count number of class instances (can't find a way to update this when an instance is deleted)
```

Comment:
hmm, I'm not understanding the need to have this counter. 
Can you explain?

Reply:
I was envisioning a system where you have a separate class instance for each run of the workflow, and then combine them later, so this would tell you how many runs of a workflow the application is considering.